airflow/api_connexion/endpoints/variable_endpoint.py

```diff
@@ -14,41 +14,76 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import Optional
 
-# TODO(mik-laj): We have to implement it.
-#     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8142
+from flask import Response, make_response, request
+from marshmallow import ValidationError
 
+from airflow.api_connexion.exceptions import BadRequest, NotFound
+from airflow.api_connexion.schemas.variable_schema import variable_collection_schema, variable_schema
+from airflow.models import Variable
+from airflow.utils.session import provide_session
 
-def delete_variable():
+
+def delete_variable(variable_key: str) -> Response:
     """
     Delete variable
     """
-    raise NotImplementedError("Not implemented yet.")
+    Variable.delete(variable_key)
+    return make_response('', 204)
 
 
-def get_variable():
+def get_variable(variable_key: str) -> Response:
     """
     Get a variables by key
     """
-    raise NotImplementedError("Not implemented yet.")
+    try:
+        var = Variable.get(variable_key)
+    except KeyError:
+        raise NotFound("Variable not found")
+    return variable_schema.dump({"key": variable_key, "val": var})
 
 
-def get_variables():
+@provide_session
+def get_variables(session, limit: Optional[int], offset: Optional[int] = None) -> Response:
     """
     Get all variable values
     """
-    raise NotImplementedError("Not implemented yet.")
+    query = session.query(Variable)
+    if limit:
+        if offset:
+            query = query.offset(offset * limit)
+        query = query.limit(limit)
+    variables = query.all()
```

Comment:
We need to get limit and offset from `request.args`
An example:
```
offset = request.args.get(parameters.page_offset, 0)
limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
```
The parameters module should have page_offset and page_limit variables defined as  equal to 'offset' and 'limit' respectively as to be consistent with what has already been merged.
The query should for limit and offset should be like this:
```
variables = query.offset(offset).limit(limit)
```

Reply:
we should just the query param passed in by connexion through the keyword arguments here. There is no need to parse the query param again manually. connexion also takes care of injecting the 100 default value from from openapi spec for you.

as for offset handling, the offset should be calculated using `offset * limit`. The connection implementation uses `query.offset(offset).limit(limit)`, which is a bug. You can port the `test_should_get_list_variables` test to connection test suite and checkout the test error.

Reply:
Thanks for pointing this out +1

Reply:
`offset * limit` is not working correctly. I didn't see any issue when I used `query.offset(offset).limit(limit)`, I suggest you add a test that `query.offset(offset).limit(limit)` cannot pass.